src/main/java/com/google/devtools/build/lib/skyframe/ProcessPackageDirectory.java

```diff
@@ -282,9 +282,11 @@ private Iterable<SkyKey> getSubdirDeps(
       }
       String basename = dirent.getName();
       PathFragment subdirectory = rootRelativePath.getRelative(basename);
-      if (!siblingRepositoryLayout && subdirectory.equals(LabelConstants.EXTERNAL_PACKAGE_NAME)) {
-        // Subpackages under //external can be processed only when
-        // --experimental_sibling_repository_layout is set.
+      if (!siblingRepositoryLayout
+          && subdirectory.equals(LabelConstants.EXTERNAL_PACKAGE_NAME)
+          && repositoryName.isMain()) {
```

Comment:
I could filter out non-Bzlmod repos here by checking for `+` in the repo name. In the current form, this commit would allow `bazel query @workspace_repo//...` to descend into `@workspace_repo//external/foo`, but targets in `@workspace_repo` could not reference anything in `@workspace_repo//external` (since `//external` will not be relative to `@workspace_repo`). I think that's okay, but I can also make this check narrower.

Reply:
no need to go that deep IMO